lib/spack/spack/util/executable.py

```diff
@@ -144,7 +144,7 @@ def streamify(arg, mode):
 
         cmd = self.exe + list(args)
 
-        cmd_line = ' '.join(cmd)
+        cmd_line = "'%s'" % "' '".join(map(lambda arg: arg.replace("'", """'"'"'"""), cmd))
```

Comment:
Instead of quoting a single quote as `"'"` (double-single-double), you can also quote it as `\'` (backslash-single). I find this easier to read (and it's also shorter). This would lead to the expression `r"\'"` (double-backslash-single-double).

Things get really fun if a quoted expression is quoted again. This quickly turns into an unholy mess that, almost magically, continues to work fine. If that's ever an issue, then we can try different strategies to quote a string, and choose the shortest one.

Alternatives to the currently chosen "enclose in single quotes" are:
- enclose in double quotes, quoting all special characters with backslashes (inside the double quotes)
- don't enclose in quotes; instead, use backslashes for unsafe characters

Really, isn't there a Python module for this, or a Python module that one could copy?